LiteCore/Storage/SQLiteKeyStore.cc

```diff
@@ -87,7 +83,7 @@ namespace litecore {
                                 "  version BLOB,"
                                 "  body BLOB,"
                                 "  extra BLOB)"));
-        _existence = db().inTransaction() ? kUncommitted : kCommitted;
+        _uncommitedTable = db().inTransaction();
```

Comment:
This corresponds to "uncommitted create table," The other part is "uncommitted delete table." Do we need to worry about it? Perhaps not, just mention it. We may probably tolerate the bias: a good Collection object is guaranteed to be good, but the bad (invalid) collection object may not be bad. User should try to get the collection object again.

Reply:
No, the reason to worry about uncommitted create table is in the case of a transaction in which a table is created and several cached statements are made.  If the transaction is aborted, those cached statements also need to be discarded since they are cached at the SQLite level and refer to a now nonexistent table.

Reply:
Likewise, if delete-table gets aborted, we will have collection object marked invalid and not reverted given that the transaction is aborted.

Reply:
I was talking about "uncommitted delete table."

Reply:
This part is not related directly to collections.  I am not worried about an aborted transaction causing collections to be falsely marked invalid since the only thing that happens is that the next time around the collection objects are recreated.  There was nothing in the logic before to handle this case and I don't see a reason to add it.